src/webgpu/shader/validation/expression/call/builtin/firstLeadingBit.spec.ts

```diff
@@ -0,0 +1,96 @@
+const builtin = 'firstLeadingBit';
+export const description = `
+Validation tests for the ${builtin}() builtin.
+`;
+
+import { makeTestGroup } from '../../../../../../common/framework/test_group.js';
+import { keysOf, objectsToRecord } from '../../../../../../common/util/data_tables.js';
+import {
+  kConcreteIntegerScalarsAndVectors,
+  kFloatScalarsAndVectors,
+} from '../../../../../util/conversion.js';
+import { ShaderValidationTest } from '../../../shader_validation_test.js';
+
+import {
+  fullRangeForType,
+  kConstantAndOverrideStages,
+  stageSupportsType,
+  validateConstOrOverrideBuiltinEval,
+} from './const_override_validation.js';
+
+export const g = makeTestGroup(ShaderValidationTest);
+
+const kValuesTypes = objectsToRecord(kConcreteIntegerScalarsAndVectors);
+
+g.test('values')
+  .desc(
+    `
+Validates that constant evaluation and override evaluation of ${builtin}() never errors
+`
+  )
+  .params(u =>
+    u
+      .combine('stage', kConstantAndOverrideStages)
+      .combine('type', keysOf(kValuesTypes))
+      .filter(u => stageSupportsType(u.stage, kValuesTypes[u.type]))
+      .beginSubcases()
+      .expand('value', u => fullRangeForType(kValuesTypes[u.type]))
+  )
+  .fn(t => {
+    const expectedResult = true; // firstLeadingBit() should never error
+    validateConstOrOverrideBuiltinEval(
+      t,
+      builtin,
+      expectedResult,
+      [kValuesTypes[t.params.type].create(t.params.value)],
+      t.params.stage
+    );
+  });
+
+const kFloatTypes = objectsToRecord(kFloatScalarsAndVectors);
+
+g.test('float_argument')
+  .desc(
+    `
+Validates that float arguments are rejected by ${builtin}()
+`
+  )
+  .params(u => u.combine('type', keysOf(kFloatTypes)))
+  .fn(t => {
+    const type = kFloatTypes[t.params.type];
+    validateConstOrOverrideBuiltinEval(
+      t,
+      builtin,
+      /* expectedResult */ false,
+      [type.create(0)],
+      'constant'
+    );
+  });
+
+const kArgCases = {
+  good: '(1u)',
+  bad_no_parens: '',
+  // Bad number of args
+  bad_too_few: '()',
+  bad_too_many: '(1u,2u)',
+  // Bad value for arg 0 (Note that float type arguments are handled in 'float_argument' above)
+  bad_0bool: '(false)',
+  bad_0array: '(array(1u))',
+  bad_0struct: '(modf(2.2))',
+};
```

Comment:
Few more to add here, textures, samplers, matrices, vector, atomic

Reply:
We haven't done those types in other similar tests.  So I'd be ok with filing a bug to followup to do this more broadly.

Also, textures and samplers can't be constructed in the middle of an expression, so we'd have to add a preamble to declare those variables.

Reply:
Atomic is another case you can't construct in an expresison. you'd need a variable declaration somewhere.

Reply:
These are the tests I've been adding as I worked down the list of builtins. You can see some examples in https://github.com/gpuweb/cts/blob/c06f419947827ece5335e4593ca466a31166c7d9/src/webgpu/shader/validation/expression/call/builtin/acos.spec.ts#L188

Reply:
Done.